slither/tools/read_storage/__main__.py

```diff
@@ -122,6 +128,12 @@ def main() -> None:
 
     srs = SlitherReadStorage(contracts, args.max_depth)
 
+    if args.block:
+        try:
+            srs.block = int(args.block)
```

Comment:
`block` should be added as a member of `SlitherReadStorage`:  https://github.com/crytic/slither/blob/f0c87149f710ed80674e2b5136bb7f6657264054/slither/tools/read_storage/read_storage.py#L54-L65

as a `Union[str, int]`. We can use `latest` as the default value